rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -416,16 +416,34 @@ class DebruijnInterpreter[M[_], F[_]](
         _env.put(addr)
       }
 
-      def addUrn(newEnv: Env[Par], urn: String): Either[ReduceError, Env[Par]] =
-        urnMap.get(urn) match {
-          case Some(p) => newEnv.put(p).asRight[ReduceError]
-          case None    => ReduceError(s"Unknown urn for new: $urn").asLeft[Env[Par]]
-        }
+      def addUrn(
+          deployParameters: DeployParameters
+      )(newEnv: Env[Par], urn: String): Either[ReduceError, Env[Par]] =
+        if (urn == "rho:deployer:auth") {
+          val bs = deployParameters.userId.exprs.headOption match {
+            case Some(Expr(GByteArray(byteString))) => byteString
+            case Some(_) | None                     => ByteString.EMPTY
```

Comment:
`case _ =>`

Reply:
this should be raising an error. We don't want the Auth-s to equal accidentally, and there's no use case for them outside of deploy (where they must be defined)

Reply:
How would they equal accidentally? `deployParamteres` is always available in deploys.
If you want to raise an error here, then we have to provide a dummy auth value for the RholangCLI and probably for a bunch of tests.

Reply:
> How would they equal accidentally

That's the thing with accidents, you never know ;)

> deployParamteres is always available in deploys.

The types don't guard that, so there are chances this will change without anybody noticing - until s/b expliots it.

> If you want to raise an error here, then we have to provide a dummy auth value for the RholangCLI and probably for a bunch of tests.

We don't, because none of them use the "rho:deployer:auth" uri.

Reply:
Yeah, makes sense.

Reply:
Done